pkg/snapshot/snapshotter/snapshotter_test.go

```diff
@@ -452,6 +460,140 @@ var _ = Describe("Snapshotter", func() {
+					GarbageCollectionPolicy:  brtypes.GarbageCollectionPolicyExponential,
+					MaxBackups:               maxBackups,
+				}
+
+				ssr, err := NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				gcCtx, cancel := context.WithTimeout(testCtx, testTimeout)
+				defer cancel()
+				ssr.RunGarbageCollector(gcCtx.Done())
+
+				list, err := store.List()
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(len(list)).Should(Equal(len(expectedSnapList)))
+
+				for index, snap := range list {
+					if snap.CreatedOn != expectedSnapList[index].CreatedOn || snap.Kind != expectedSnapList[index].Kind || snap.SnapDir != expectedSnapList[index].SnapDir {
+						Fail("Expected snap list doesn't match with output snap list")
+					}
+				}
+			})
+
+			//Test to check backward compatibility of garbage collector
+			//Tests garbage collector behaviour (in exponential config) when only v1 directory is present
+			//TODO: Consider removing when backward compatibility no longer needed
+			It("should garbage collect exponentially with only v1 dir structure present (backward compatible test)", func() {
+				logger.Infoln("creating expected output")
+
+				// Prepare expected resultant snapshot list
+				var (
+					now = time.Now().UTC()
+					//store            = prepareBackwardCompatibleStoreInV1ForGC(now, "gc_exponential_backward_compatiblev1.bkp")
+					store            = prepareStoreForGarbageCollectionInPrefix(now, "gc_exponential_backward_compatiblev1.bkp", "v1")
+					snapTime         = time.Date(now.Year(), now.Month(), now.Day()-35, 0, -30, 0, 0, now.Location())
+					expectedSnapList = brtypes.SnapList{}
+				)
+
+				expectedSnapList = prepareExpectedSnapshotsList(snapTime, now, snapsInV1)
+
+				//start test
+				snapshotterConfig := &brtypes.SnapshotterConfig{
+					FullSnapshotSchedule:     schedule,
+					DeltaSnapshotPeriod:      wrappers.Duration{Duration: 10 * time.Second},
+					DeltaSnapshotMemoryLimit: brtypes.DefaultDeltaSnapMemoryLimit,
+					GarbageCollectionPeriod:  wrappers.Duration{Duration: garbageCollectionPeriod},
+					GarbageCollectionPolicy:  brtypes.GarbageCollectionPolicyExponential,
+					MaxBackups:               maxBackups,
+				}
+
+				ssr, err := NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				gcCtx, cancel := context.WithTimeout(testCtx, testTimeout)
+				defer cancel()
+				ssr.RunGarbageCollector(gcCtx.Done())
+
+				list, err := store.List()
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(len(list)).Should(Equal(len(expectedSnapList)))
+
+				for index, snap := range list {
+					if snap.CreatedOn != expectedSnapList[index].CreatedOn || snap.Kind != expectedSnapList[index].Kind || snap.SnapDir != expectedSnapList[index].SnapDir {
+						Fail("Expected snap list doesn't match with output snap list")
+					}
```

Comment:
Might be better to change this section to 3 `Expect`s for the corresponding fields. This will help debug when the tests fail.

The same holds in other cases below.

Reply:
Done